core/src/main/java/io/grpc/ClientCallImpl.java

```diff
@@ -85,27 +108,21 @@
     ClientTransport get();
   }
 
+  private enum State {INIT, STARTED, HALF_CLOSED, CANCELED}
+
   ClientCallImpl<ReqT, RespT> setUserAgent(String userAgent) {
     this.userAgent = userAgent;
     return this;
   }
 
   @Override
-  public void start(Listener<RespT> observer, Metadata.Headers headers) {
-    Preconditions.checkState(stream == null, "Already started");
+  public void start(ClientCall.Listener<RespT> observer, Metadata.Headers headers) {
```

Comment:
I had done that at one point, but the inner class ClientStreamListenerImpl also wanted to use that name.   (I agree, those names are better!)  However, based on the comments of the previous few PRs, I was intentionally trying to change as little as possible here.  (I would be delighted to make those name changes, if you are alright with it)